pymeasure/instruments/lecroy/lecroyT3DSO1204.py

```diff
@@ -0,0 +1,1324 @@
+    return output
+
+
+def _set_measure_parameter_validator(value, values):
+    """ Validate the input of the set_measure_parameter property
+    :param value: input parameters as a 2-element tuple
+    :param values: allowed space for each parameter """
+    if not isinstance(value, tuple):
+        raise ValueError('Input value {} of trigger_select should be a tuple'.format(value))
+    if len(value) != 2:
+        raise ValueError('Number of parameters {} different from 2'.format(len(value)))
+    output = (value[0], sanitize_source(value[1]))
+    for i in range(2):
+        strict_discrete_set(output[i], values=values[i])
+    return output
+
+
+def _intensity_validator(value, values):
+    """ Validate the input of the intensity property (grid intensity and trace intensity)
+    :param value: input parameters as a 2-element tuple
+    :param values: allowed space for each parameter """
+    if not isinstance(value, tuple):
+        raise ValueError('Input value {} of trigger_select should be a tuple'.format(value))
+    if len(value) != 2:
+        raise ValueError('Number of parameters {} different from 2'.format(len(value)))
+    for i in range(2):
+        strict_discrete_range(value=value[i], values=values[i], step=1)
+    return value
+
+
+class _ChunkResizer:
+    """ The only purpose of this class is to resize the chunk size of the instrument adapter.
+    This is necessary when reading a big chunk of data from the oscilloscope like image dumps and
+    waveforms.
+    Note.
+    Only if the new chunk size is bigger than the current chunk size, it is resized. """
+
+    def __init__(self, adapter, chunk_size):
+        """ Just initialize the object attributes.
+        :param: adapter of the instrument. This is usually accessed through the
+        Instrument::adapter attribute.
+        :chunk_size: new chunk size (int).
+        """
+        self.adapter = adapter
+        self.old_chunk_size = None
+        self.new_chunk_size = int(chunk_size) if chunk_size else 0
+
+    def __enter__(self):
+        """ Only resize the chunk size if the adapter support this feature"""
+        if (self.adapter.connection is not None
+                and hasattr(self.adapter.connection, "chunk_size")):
+            if self.new_chunk_size > self.adapter.connection.chunk_size:
+                self.old_chunk_size = self.adapter.connection.chunk_size
+                self.adapter.connection.chunk_size = self.new_chunk_size
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if self.old_chunk_size is not None:
+            self.adapter.connection.chunk_size = self.old_chunk_size
+
+
+# noinspection DuplicatedCode
+class Channel:
+    """ Implementation of a LeCroy T3DSO1204 Oscilloscope channel.
+
+    Implementation modeled on Channel object of Keysight DSOX1102G instrument. """
+
+    _BOOLS = {True: "ON", False: "OFF"}
+
+    def __init__(self, instrument, number):
+        self.instrument = instrument
+        self.number = number
```

Comment:
If you use the Channel implementation, that part would be superfluous.

For your information, the names would be `parent` instead of `instrument` and `id` instead of `number`.

Reply:
Same as the previous comment.

Reply:
Done